providers/manage.rb

```diff
@@ -111,10 +111,12 @@ def chef_solo_search_installed?
       if manage_home_files?(home_dir, u['username'])
         Chef::Log.debug("Managing home files for #{u['username']}")
 
-        directory "#{home_dir}/.ssh" do
-          owner u['username']
-          group u['gid'] || u['username']
-          mode "0700"
+        if u['ssh_keys'] || u['ssh_private_key'] || ['ssh_public_key']
```

Comment:
I've patched the "u['ssh_public_key']" as you suggested, thank you.

I'm having some difficulty integrating tests for this: ideally, it would require writing to a user's home directory, and I admit I've *not* been writing the kind of test suite suggested. Can I dig into that later, perhaps? The insistence on creating $HOME/.ssh is seriously breaking this tool in root-squashed NFS environments.